lib/rubocop/cop/correctors/alignment_corrector.rb

```diff
@@ -39,27 +41,42 @@ def align_end(processed_source, node, align_to)
         private
 
         def autocorrect_line(corrector, line_begin_pos, expr, column_delta,
-                             heredoc_ranges)
+                             taboo_ranges)
           range = calculate_range(expr, line_begin_pos, column_delta)
-          # We must not change indentation of heredoc strings.
-          return if heredoc_ranges.any? { |h| within?(range, h) }
+          # We must not change indentation of heredoc strings or inside other
+          # string literals
+          return if taboo_ranges.any? { |t| within?(range, t) }
 
           if column_delta.positive?
-            unless range.source == "\n"
-              # TODO: Fix ranges instead of using `begin`
-              corrector.insert_before(range.begin, ' ' * column_delta)
+            unless range.resize(1).source == "\n"
+              corrector.insert_before(range, ' ' * column_delta)
             end
           elsif range.source =~ /\A[ \t]+\z/
             remove(range, corrector)
           end
         end
 
-        def heredoc_ranges(node)
+        def inside_string_ranges(node)
           return [] unless node.is_a?(Parser::AST::Node)
 
-          node.each_node(:dstr)
-              .select(&:heredoc?)
-              .map { |n| n.loc.heredoc_body.join(n.loc.heredoc_end) }
+          node.each_node(:str, :dstr, :xstr).map { |n| inside_string_range(n) }
+              .compact
+        end
+
+        def inside_string_range(node)
+          if node.heredoc?
+            node.loc.heredoc_body.join(node.loc.heredoc_end)
+          else
+            inside_regular_string_range(node)
+          end
+        end
+
+        def inside_regular_string_range(node)
+          loc = node.location
+          return unless loc.respond_to?(:begin) && loc.respond_to?(:end)
```

Comment:
Would be cool to extract these two guard clause conditions into a named method. WDYT?

Reply:
Sounds good. After extracting the guard clause, the remaining method begged me to be inlined. 😉 I also added some comments.